include/irr/asset/IMeshPacker.h

```diff
@@ -0,0 +1,36 @@
+#ifndef __IRR_I_MESH_PACKER_H_INCLUDED__
+#define __IRR_I_MESH_PACKER_H_INCLUDED__
+#include <cstdint>
+#include <optional>
+
+namespace irr
+{
+namespace asset
+{
+//where should I move it?
+struct DrawElementsIndirectCommand_t
+{
+    uint32_t count;
+    uint32_t instanceCount;
+    uint32_t firstIndex;
+    uint32_t baseVertex;
+    uint32_t baseInstance;
+};
+
+template <typename MeshType>
+class IMeshPacker
+{
+public:
+    virtual std::optional<std::pair<MeshType* ,DrawElementsIndirectCommand_t>> packMeshes(core::vector<MeshType*>& meshes) = 0;
```

Comment:
use iterators begin and end, much more flexible than vector

Reply:
pack meshbuffers not meshes.

Also split the business logic into `allocate` and `commit`.

One figures out how much space will need to be reserved, reserves it etc., returns some useful metadata for a later `commit`etc. etc.

`commit` actually does the heavy lifting.